lib/lotus/repository.rb

```diff
@@ -551,6 +551,42 @@ def transaction(options = {})
         end
       end
 
+
+      # Executes the given raw sql on the adapter.
+      #
+      # Please note that it's only supported by some databases, a `NotImplementedError` will be
+      # raised when the adapter does not responds to the `execute` method.
+      #
+      # For advanced scenarios, please check the documentation of each adapter.
+      #
+      # @see Lotus::Model::Adapters::SqlAdapter#execute
+      #
+      # @since
+      #
+      # @example Basic usage with SQL adapter
+      #   require 'lotus/model'
+      #
+      #   class Article
+      #     include Lotus::Entity
+      #     attributes :title, :body
+      #   end
+      #
+      #   class ArticleRepository
+      #     include Lotus::Repository
+      #   end
+      #
+      #   article = Article.new(title: 'Introducing transactions',
+      #     body: 'lorem ipsum')
+      #
+      #   resultset = ArticleRepository.execute("select * from articles")
+      #   resultset.each_hash do |result|
+      #     result # -> { id: 123, title: "Introducing transactions", body: "lorem ipsum"}
+      #   end
+      def execute(sql)
```

Comment:
@peterberkenbosch Probably `raw` is a better argument name. Please remember that at this level we are targeting any potential storage, so `sql` is a too specific term. :smile: